src/binder/clrprivbinderassemblyloadcontext.cpp

```diff
@@ -160,7 +159,8 @@ HRESULT CLRPrivBinderAssemblyLoadContext::BindUsingPEImage( /* in */ PEImage *pP
             {
                 // The simple name of the assembly being requested to be bound was found in the TPA list.
                 // Now, perform the actual bind to see if the assembly was really in the TPA assembly or not.
-                hr = m_pTPABinder->BindAssemblyByNameWorker(pAssemblyName, &pCoreCLRFoundAssembly);
+                // Don't search app paths when using the TPA binder because the actual binder is using a host assembly resolver.
+                hr = m_pTPABinder->BindAssemblyByNameWorker(pAssemblyName, &pCoreCLRFoundAssembly, true /* excludeAppPaths */);
```

Comment:
@sergiy-k found that I was passing in the wrong value for the parameter here. Thanks for finding this, fixed.